### PR TITLE
ACP-77: Add caching to SoV DB helpers

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

Factored out of #3388

## How this works

Adds caching to `getSubnetOnlyValidator`, `putSubnetOnlyValidator`, and `deleteSubnetOnlyValidator`.

## How this was tested

Modified unit test

## Need to be documented in RELEASES.md?

no

### CODE REVIEW

File: vms/platformvm/state/subnet_only_validator.go [hunk 1]
```diff
@@ -103,7 +107,18 @@ func (v SubnetOnlyValidator) immutableFieldsAreUnmodified(o SubnetOnlyValidator)
...
+	db database.KeyValueReader,
+	validationID ids.ID,
+) (SubnetOnlyValidator, error) {
+	if maybeSOV, ok := cache.Get(validationID); ok {
+		if maybeSOV.IsNothing() {
+			return SubnetOnlyValidator{}, database.ErrNotFound
+		}
+		return maybeSOV.Value(), nil
+	}
+
```
Comment (ceyonur) [inline vms/platformvm/state/subnet_only_validator.go:121]: should we cache the sov here if it's not found in cache but in DB?

File: vms/platformvm/state/subnet_only_validator.go [hunk 2]
```diff
@@ -118,14 +133,32 @@ func getSubnetOnlyValidator(db database.KeyValueReader, validationID ids.ID) (Su
...
+func deleteSubnetOnlyValidator(
+	db database.KeyValueDeleter,
+	cache cache.Cacher[ids.ID, maybe.Maybe[SubnetOnlyValidator]],
+	validationID ids.ID,
+) error {
+	if err := db.Delete(validationID[:]); err != nil {
+		return err
+	}
+
+	cache.Put(validationID, maybe.Nothing[SubnetOnlyValidator]())
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/state/subnet_only_validator.go:162]: `Put` caches that the entry is not on disk. `Evict` would remove any cached entry.

File: vms/platformvm/state/subnet_only_validator.go (hunk 1 above)
Comment (StephenButtolph) [reply vms/platformvm/state/subnet_only_validator.go:119]: added caching here

File: vms/platformvm/state/subnet_only_validator_test.go [hunk 3]
```diff
@@ -139,11 +141,8 @@ func TestSubnetOnlyValidator_immutableFieldsAreUnmodified(t *testing.T) {
 }
 
 func TestSubnetOnlyValidator_DatabaseHelpers(t *testing.T) {
-	require := require.New(t)
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/platformvm/state/subnet_only_validator_test.go:142]: done

**Review by ceyonur — APPROVED** [review]

**Pull request merged**